packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -953,19 +956,26 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
     assert(() {
       // Only set this if it's null to save work. It gets reset to null if the
       // _direction changes.
-      final String debugOverflowHints =
-        'The overflowing $runtimeType has an orientation of $_direction.\n'
-        'The edge of the $runtimeType that is overflowing has been marked '
-        'in the rendering with a yellow and black striped pattern. This is '
-        'usually caused by the contents being too big for the $runtimeType. '
-        'Consider applying a flex factor (e.g. using an Expanded widget) to '
-        'force the children of the $runtimeType to fit within the available '
-        'space instead of being sized to their natural size.\n'
-        'This is considered an error condition because it indicates that there '
-        'is content that cannot be seen. If the content is legitimately bigger '
-        'than the available space, consider clipping it with a ClipRect widget '
-        'before putting it in the flex, or using a scrollable container rather '
-        'than a Flex, like a ListView.';
+      final List<DiagnosticsNode> debugOverflowHints = <DiagnosticsNode>[
+        ErrorDescription(
+          'The overflowing $runtimeType has an orientation of $_direction.'
+        ),
+        ErrorHint(
+          'The edge of the $runtimeType that is overflowing has been marked '
+          'in the rendering with a yellow and black striped pattern. This is '
+          'usually caused by the contents being too big for the $runtimeType. '
+          'Consider applying a flex factor (e.g. using an Expanded widget) to '
```

Comment:
The hint starts from this line. I'd consider the three lines above it a description of the error.

Reply:
done